boa/src/vm/mod.rs

```diff
@@ -601,11 +832,7 @@ impl Context {
         const OPERAND_COLUMN_WIDTH: usize = COLUMN_WIDTH;
         const NUMBER_OF_COLUMNS: usize = 4;
 
-        let msg = if self.vm.frame().exit_on_return {
-            " VM Start"
-        } else {
-            " Call Frame "
-        };
+        let msg = " VM Start";
```

Comment:
Is there a reason why you changed this?
The output shows "VM Start" twice when running because a CodeBlock also shows as "VM Start". 

This logic was to distinguish between the main VM running and a codeblock being executed

Reply:
Fixed the output to show `Call Frame` and `VM Start` again. That just got lost when I removed `exit_on_return`.

Can you show an example how you think the output should look for the two other points? You mean just like this?:

```
-------------------------------------- VM Start---------------------------------------

Time         Opcode                   Operands                 Top Of Stack
1μs          PushLiteral              0                        "hello"
```

Reply:
I would expect something like this:
```powershell
------------------------------ Compiled Output ----------------------------------------
    Location  Count   Opcode              Operands

    000000    0000    GetFunction         0000: '' (length: 0)
    000005    0001    DefInitLet          0000: 'o'
    000010    0002    PushEmptyObject     
    000011    0003    Dup                 
    000012    0004    GetFunction         0001: '' (length: 0)
    000017    0005    Swap                
    000018    0006    SetPropertyByName   0001: 'get'
    000023    0007    PushLiteral         0
    000028    0008    GetName             0000: 'o'
    000033    0009    GetName             0002: 'Object'
    000038    0010    Dup                 
    000039    0011    GetPropertyByName   0003: 'defineProperty'
    000044    0012    Call                3
    000049    0013    Pop                 
    000050    0014    GetName             0000: 'o'
    000055    0015    GetPropertyByName   0004: 'test'

Literals:
    0000: <string> "test"

Names:
    0000: o
    0001: get
    0002: Object
    0003: defineProperty
    0004: test

Functions:
    0000: name: '' (length: 0)
    0001: name: '' (length: 0)


-------------------------------------- VM Start---------------------------------------
Time         Opcode                   Operands                 Top Of Stack

42μs         GetFunction              0000: '' (length: 0)     [function]
15μs         DefInitLet               0000: 'o'                <empty>
5μs          PushEmptyObject                                   [object]
1μs          Dup                                               [object]
47μs         GetFunction              0001: '' (length: 0)     [function]
0μs          Swap                                              [object]
19μs         SetPropertyByName        0001: 'get'              [object]
1μs          PushLiteral              0                        "test"
6μs          GetName                  0000: 'o'                [function]
20μs         GetName                  0002: 'Object'           [function]
1μs          Dup                                               [function]
6μs          GetPropertyByName        0003: 'defineProperty'   [function]
45μs         Call                     3                        [function]
1μs          Pop                                               <empty>
5μs          GetName                  0000: 'o'                [function]

----------------- Call Frame '' [Func Identifier from above]---------------------------------------------
    Location  Count   Opcode              Operands

    000000    0000    PushLiteral         0
    000005    0001    Return              
    000006    0002    PushUndefined       
    000007    0003    Return              

Literals:
    0000: <string> "hello"

Names:
    <empty>

Functions:
    <empty>


-------------------------------------- Call Frame '' [Func Identifier from above]------------------------
Time         Opcode                   Operands                 Top Of Stack

1μs          PushLiteral              0                        "hello"
2μs          Return                                            "hello"
95μs         GetPropertyByName        0004: 'test'             "hello"

Stack:
0000         "hello"

"hello"
```

- The first section called "Compiled Output" or something along those lines, as its not ran at this point this is just the code generated. 
- A space between the headings and the output to make it easier to read.
- `name '' (length: 0)` being changed to `Call Frame` (do we care about the length?) If it can reference the function number its executing from that would be even more useful. so `Call Frame [001]` or something

Reply:
I implemented the empty lines and the switch to `Compiled Output`.

The functions ID's and the confusing output after a Call Frame returns require some more thought. I will open a issue for those, as I cannot think of a good way to solve this right now.